File: app/src/androidTest/java/com/owncloud/android/ui/dialog/SyncFileNotEnoughSpaceDialogFragmentTest.java
/*
 *
 * Nextcloud Android client application
 *
 * @author Tobias Kaminsky
 * Copyright (C) 2020 Tobias Kaminsky
 * Copyright (C) 2020 Nextcloud GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

package com.owncloud.android.ui.dialog;

import android.content.Intent;

import com.owncloud.android.AbstractIT;
import com.owncloud.android.datamodel.OCFile;
import com.owncloud.android.ui.activity.FileDisplayActivity;
import com.owncloud.android.utils.ScreenshotTest;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import java.util.Objects;

import androidx.test.core.app.ActivityScenario;
import androidx.test.core.app.ApplicationProvider;

public class SyncFileNotEnoughSpaceDialogFragmentTest extends AbstractIT {
    private ActivityScenario<FileDisplayActivity> scenario;

    @Before
    public void setUp() {
        Intent intent = new Intent(ApplicationProvider.getApplicationContext(), FileDisplayActivity.class);
        scenario = ActivityScenario.launch(intent);
    }

    @After
    public void tearDown() {
        scenario.close();
    }

    @Test
    @ScreenshotTest
    public void showNotEnoughSpaceDialogForFolder() {
        scenario.onActivity(test -> {
            OCFile ocFile = new OCFile("/Document/");
            ocFile.setFileLength(5000000);
            ocFile.setFolder();

            SyncFileNotEnoughSpaceDialogFragment dialog = SyncFileNotEnoughSpaceDialogFragment.newInstance(ocFile, 1000);
            dialog.show(test.getListOfFilesFragment().getFragmentManager(), "1");

            onIdleSync(() -> screenshot(Objects.requireNonNull(dialog.requireDialog().getWindow()).getDecorView()));
        });
    }

    @Test
    @ScreenshotTest
    public void showNotEnoughSpaceDialogForFile() {
        scenario.onActivity(test -> {
            OCFile ocFile = new OCFile("/Video.mp4");
            ocFile.setFileLength(1000000);

            SyncFileNotEnoughSpaceDialogFragment dialog = SyncFileNotEnoughSpaceDialogFragment.newInstance(ocFile, 2000);
            dialog.show(test.getListOfFilesFragment().getFragmentManager(), "2");

           onIdleSync(() -> screenshot(Objects.requireNonNull(dialog.requireDialog().getWindow()).getDecorView()));
        });
    }
}

File: app/src/androidTest/java/com/owncloud/android/ui/dialog/SyncFileNotEnoughSpaceDialogFragmentTest.kt
/*
 * Nextcloud - Android Client
 *
 * SPDX-FileCopyrightText: 2024 Alper Ozturk <alper.ozturk@nextcloud.com>
 * SPDX-License-Identifier: AGPL-3.0-or-later
 */
package com.owncloud.android.ui.dialog

import androidx.test.core.app.launchActivity
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.IdlingRegistry
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import androidx.test.espresso.matcher.ViewMatchers.isRoot
import com.owncloud.android.AbstractIT
import com.owncloud.android.datamodel.OCFile
import com.owncloud.android.ui.activity.FileDisplayActivity
import com.owncloud.android.ui.dialog.SyncFileNotEnoughSpaceDialogFragment.Companion.newInstance
import com.owncloud.android.utils.EspressoIdlingResource
import com.owncloud.android.utils.ScreenshotTest
import org.junit.After
import org.junit.Before
import org.junit.Test

class SyncFileNotEnoughSpaceDialogFragmentTest : AbstractIT() {
    private val testClassName = "com.owncloud.android.ui.dialog.SyncFileNotEnoughSpaceDialogFragmentTest"

    @Before
    fun registerIdlingResource() {
        IdlingRegistry.getInstance().register(EspressoIdlingResource.countingIdlingResource)
    }

    @After
    fun unregisterIdlingResource() {
        IdlingRegistry.getInstance().unregister(EspressoIdlingResource.countingIdlingResource)
    }

    @Test
    @ScreenshotTest
    fun showNotEnoughSpaceDialogForFolder() {
        launchActivity<FileDisplayActivity>().use { scenario ->
            scenario.onActivity { sut ->
                val ocFile = OCFile("/Document/").apply {
                    fileLength = 5000000
                    setFolder()
                }

                onIdleSync {
                    EspressoIdlingResource.increment()
                    newInstance(ocFile, 1000).apply {
                        show(sut.supportFragmentManager, "1")
                    }
                    EspressoIdlingResource.decrement()

                    val screenShotName = createName(testClassName + "_" + "showNotEnoughSpaceDialogForFolder", "")
                    onView(isRoot()).check(matches(isDisplayed()))
                    screenshotViaName(sut, screenShotName)
                }
            }
        }
    }

    @Test
    @ScreenshotTest
    fun showNotEnoughSpaceDialogForFile() {
        launchActivity<FileDisplayActivity>().use { scenario ->
            scenario.onActivity { sut ->
                val ocFile = OCFile("/Video.mp4").apply {
                    fileLength = 1000000
                }

                onIdleSync {
                    EspressoIdlingResource.increment()
                    newInstance(ocFile, 2000).apply {
                        show(sut.supportFragmentManager, "2")
                    }
                    EspressoIdlingResource.decrement()

                    val screenShotName = createName(testClassName + "_" + "showNotEnoughSpaceDialogForFile", "")
                    onView(isRoot()).check(matches(isDisplayed()))
                    screenshotViaName(sut, screenShotName)
                }
            }
        }
    }
}
